warehouse/query-core/src/main/java/datawave/query/util/sortedmap/HdfsBackedSortedMap.java

```diff
@@ -0,0 +1,302 @@
+package datawave.query.util.sortedmap;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FsStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.Logger;
+
+import datawave.query.iterator.ivarator.IvaratorCacheDir;
+import datawave.query.iterator.ivarator.IvaratorCacheDirConfig;
+import datawave.query.util.sortedset.FileSortedSet;
+
+/**
+ * Sorted map backed by HDFS
+ *
+ * @param <K>
+ *            key of the map
+ * @param <V>
+ *            value of the map
+ */
+public class HdfsBackedSortedMap<K,V> extends BufferedFileBackedSortedMap<K,V> {
+    private static final Logger log = Logger.getLogger(HdfsBackedSortedMap.class);
+    private static final String FILENAME_PREFIX = "SortedMapFile.";
+
+    public static class Builder<B extends Builder<B,K,V>,K,V> extends BufferedFileBackedSortedMap.Builder<B,K,V> {
+        private List<IvaratorCacheDir> ivaratorCacheDirs;
+        private String uniqueSubPath;
+        private FileSortedSet.PersistOptions persistOptions;
+
+        public Builder() {
+            // change the default buffer persist threshold
+            withBufferPersistThreshold(10000);
```

Comment:
Do we really want to buffer 10,000 elements before persisting to disk? Is this configurable? Should this persist threshold be different from a standard ivarator? I.e., what if I want to buffer 5k ivarator keys but only 250 unique documents?

Small nit, would prefer the code read like 
```java
withBufferPersistThreshold(10_000);
```

Reply:
hmm, I thought this was configurable. I will fix this if that is not the case.  I expect that 10K is probably not the best default for the most recent unique functionality.  Also I like 10_000.... I will do that.

Reply:
Changed the constant to add the '_'.  This is being overridden in the UniqueTransform where the maps are built.